ge07_team01/models/motorcycle_registry.py

```diff
@@ -0,0 +1,32 @@
+from odoo import api, models, fields
+from odoo.exceptions import ValidationError
+
+
+class MotorcycleRegistry(models.Model):
+    _inherit = "motorcycle.registry"
+
+    lot_ids = fields.One2many("stock.lot", "registry_id", string="Lot IDs")
+    lot_id = fields.Many2one("stock.lot", string="Lot ID", compute="_compute_lot_id", store=True)
+    sale_order_id = fields.Many2one("sale.order")
+
+    vin = fields.Char(string="VIN", related="lot_id.name", required=False)
+    owner_id = fields.Many2one("res.partner", string="Owner", related="sale_order_id.partner_id")
+    registry_date = fields.Date(string='Registration Date', default=fields.Date.today())
+
+    @api.constrains("lot_ids")
+    def _check_lot_ids(self):
+        for registry in self:
+            if len(registry.lot_ids) > 1:
+                raise ValidationError("lot_ids cannot have more than one lot.")
+
+    @api.depends("lot_ids")
+    def _compute_lot_id(self):
+        self.lot_id = False
+        for registry in self.filtered(lambda r: r.lot_ids is not False and len(r.lot_ids) > 0):
```

Comment:
You do not need the 'is not False'

Reply:
The thing is that some registries don't have the field 'lot_ids' set. In this case the value remains as 'False', so if we try to do 'len(r.lot_ids)' we would receive an error (since it's a boolean field).

Reply:
Yeah, I meant that you can have `r.lot_ids` to make the code somewhat shorter and easy to read but the condition is needed

`self.filetered(lambda r: r.lot_ids and len(.........`